### PR TITLE
Set-Cookie support in http

### PR DESCRIPTION
**Pull request by mhoeger**
Protobuf PR: Azure/azure-functions-language-worker-protobuf#32
Host PR: https://github.com/Azure/azure-functions-host/pull/4317

For issue: https://github.com/Azure/azure-functions-host/issues/2737
Resolves: https://github.com/Azure/azure-functions-nodejs-worker/issues/222

### CODE REVIEW

File: src/WorkerChannel.ts [hunk 1]
```diff
@@ -139,7 +139,8 @@ export class WorkerChannel implements IWorkerChannel {
       }
       if (result) {
         if (result.return) {
-          response.returnValue = toTypedData(result.return);
+          let returnBinding = info.getReturnBinding();
+          response.returnValue = returnBinding ? returnBinding.converter(result.return) : toTypedData(result.return);
```
Comment (pragnagopa) [inline src/WorkerChannel.ts]: Add test verify this code path.

File: src/converters/CommonConverters.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import { 
+    AzureFunctionsRpcMessages as rpc,
+    INullableString,
+    INullableBool,
+    INullableDouble,
+    INullableTimestamp
+} from '../../azure-functions-language-worker-protobuf/src/rpc';
+
+export function fromTypedData(typedData?: rpc.ITypedData, convertStringToJson: boolean = true) {
```
Comment (pragnagopa) [inline src/converters/CommonConverters.ts]: Nit: Rename this file to rpcConverters?

File: azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto [hunk 3]
```diff
@@ -375,6 +376,44 @@ message RpcException {
   string message = 2;
 }
 
+// Http cookie type. Note that only name and value are used for Http requests
+message RpcHttpCookie {
```
Comment (pragnagopa) [inline azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto]: Nit: add link to RFC: https://www.rfc-editor.org/rfc/rfc6265.txt

File: src/converters/RpcHttpConverters.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    let rpcSameSite: rpc.RpcHttpCookie.SameSite = rpc.RpcHttpCookie.SameSite.None;
+    if (inputCookie.sameSite) {
+        if (inputCookie.sameSite.toLocaleLowerCase() === "lax") {
+        rpcSameSite = rpc.RpcHttpCookie.SameSite.Lax;
+        } else if (inputCookie.sameSite.toLocaleLowerCase() === "strict") {
+        rpcSameSite = rpc.RpcHttpCookie.SameSite.Strict;
+        }
+    }
+
+    const rpcCookie: rpc.IRpcHttpCookie = {
```
Comment (pragnagopa) [inline src/converters/RpcHttpConverters.ts]: Nit: Add link: https://www.rfc-editor.org/rfc/rfc6265.txt in the comments

File: src/http/Response.ts [hunk 5]
```diff
@@ -15,6 +18,7 @@ interface IResponse {
 export class Response implements IResponse {
     statusCode?: string | number;
     headers: {[key:string]: any} = {};
+    cookies: Cookie[] = [];
```
Comment (pragnagopa) [inline src/http/Response.ts]: Add test

File: test/InterfacesTest.ts [hunk 6]
```diff
@@ -50,6 +50,20 @@ const runFunction: AzureFunction = async function(context: Context) {
...
+const cookieFunction: AzureFunction = async function(context: Context) {
+    let cookies: Cookie[] = [
+        {
+            name: "cookiename",
+            value: "cookievalue",
+            expires: Date.now()
+        }
+    ];
+    context.res = {
+        cookies,
```
Comment (pragnagopa) [inline test/InterfacesTest.ts]: - update test to add multiple cookies
- update test to verify cookies are set on the response.

**Comment by pragnagopa** [comment]
Please add a E2E test.

**Comment by mhoeger** [comment]
@pragnagopa - as discussed offline, i've manually verified these changes. opening a separate work item for e2e tests so that we can get the changes out the upcoming release

File: azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto [hunk 7]
```diff
@@ -11,6 +11,7 @@ package AzureFunctionsRpcMessages;
 
 import "google/protobuf/duration.proto";
 import "identity/ClaimsIdentityRpc.proto";
+import "shared/NullableTypes.proto";
```
Comment (pragnagopa) [inline azure-functions-language-worker-protobuf/src/proto/FunctionRpc.proto]: Get the latest proto: https://github.com/Azure/azure-functions-language-worker-protobuf/releases/tag/v1.2.0-protofile

**Comment by mhoeger** [comment]
note: e2e tests in devops temporarily failing for core tools 2.7.1385

**Pull request merged**